Validate connection to CDP and s3a

kjmccarthy opened the pull request:
- Renamed Sentry to Role and SentryRole to SecurityRole
- Removed HDFS tests
- HiveClientImplIntegrationSpec tests succeed
- RoleClientIntegrationSpec has some tests succeed and other fail due to AD and syntax

afoerster committed:
Test outline

afoerster committed:
Disable kerberos for hive tests

kjmccarthy committed:
Validate connection to CDP and s3a

- Renamed Sentry to Role and SentryRole to SecurityRole
- Removed HDFS tests
- HiveClientImplIntegrationSpec tests succeed
- RoleClientIntegrationSpec has some tests succeed and other fail due to AD and syntax

afoerster commented on common/src/main/scala/io/phdata/clients/HiveClient.scala:
```diff
@@ -59,28 +57,21 @@ class HiveClientImpl[F[_]](loginContextProvider: LoginContextProvider, transacto
-      (sql"""SHOW TABLES in """ ++ Fragment.const(name)).query[String].to[List].transact(transactor).map { tables =>
-        HiveDatabase(name, tables.map(HiveTable.apply))
-      }
+    (sql"""SHOW TABLES in """ ++ Fragment.const(name)).query[String].to[List].transact(transactor).map { tables =>
```
@kjmccarthy did you run `sbt scalafmt`?

kjmccarthy replied:
yes, I did

afoerster commented on integration-test/src/test/scala/io/phdata/clients/RoleClientIntegrationSpec.scala:
```diff
@@ -0,0 +1,210 @@
+      result should contain(role)
+    )
+
+    testRoles.foreach(role =>
```
This should either go in an `after` method or in a `finally` block so everything gets cleaned up, even on exception

kjmccarthy replied:
This just validates the role. This block starting on line 35 cleans them up
```
  override def afterAll() {
    val cleanUp = for {
     roles <- roleClient.roles
     _ <- roles.filter(_.startsWith("arch_test_role_")).map(testRole => roleClient.dropRole(testRole)).pure[IO]
    } yield ()

    cleanUp.unsafeRunSync()
  }
```

afoerster commented on integration-test/src/test/scala/io/phdata/clients/RoleClientIntegrationSpec.scala:
```diff
@@ -0,0 +1,210 @@
+
+  it should "list roles" in {
+    val testRoles = List(
+      s"test_role_${UUID.randomUUID().toString.take(8)}",
```
do all roles in this class start with`test_role`? It's good if they do, makes it easier to clean up if something goes wrong. Maybe better, make it `arch_test_role`, then all the objects we create start with `arch_test` and we can easily recognize them

afoerster commented on integration-test/src/test/scala/io/phdata/clients/RoleClientIntegrationSpec.scala:
```diff
@@ -0,0 +1,210 @@
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+import cats.implicits._
+
+class RoleClientIntegrationSpec extends FlatSpec with Matchers with HiveTest with IntegrationTest with BeforeAndAfterAll{
```
Are all tests in this class passing? Can you note with a comment any that are not?

kjmccarthy replied:
No not all are passing. I will add a note on which aren't. There is a hive ranger command grid. 
https://cwiki.apache.org/confluence/display/RANGER/Hive+Commands+to+Ranger+Permission+Mapping

afoerster commented on integration-test/src/test/scala/io/phdata/services/JDBCConnectionIntegrationSpec.scala:
```diff
@@ -0,0 +1,19 @@
+
+import org.scalatest.{FlatSpec, Matchers}
+
+class JDBCConnectionIntegrationSpec extends FlatSpec with Matchers {
```
This can be removed since you are testing these things elsewhere

afoerster commented on provisioning/src/main/scala/io/phdata/provisioning/package.scala:
```diff
@@ -43,6 +43,6 @@ package object provisioning
   case class LocationGrant(id: Long, roleName: String, location: String)
 
-  case class SentryRole(id: Long, name: String)
+  case class SecurityRole(id: Long, name: String)
```
Nice.

kjmccarthy committed:
Updates for PR

- Add exclusion to hive-jdbc for org.apache.logging
- Changed test_role to arch_test_role
- Added comment for failed RoleClientIntegrationSpec tests
